examples/medplum-provider/src/components/encounter/EncounterChart.tsx

```diff
@@ -31,10 +38,21 @@ const TASK_COMPLETED_STATUSES = new Set<Task['status']>([
   'entered-in-error',
 ]);
 
-export const EncounterChart = (): JSX.Element => {
-  const { patientId, encounterId } = useParams();
+export interface EncounterChartProps {
+  encounter: Encounter | Reference<Encounter>;
+}
+
+export const EncounterChart = (props: EncounterChartProps): JSX.Element => {
+  const { encounter: encounterProp } = props;
   const medplum = useMedplum();
-  const patient = usePatient();
+
+  // Resolve encounter reference to actual resource
+  const encounterResource = useResource(encounterProp);
+
+  // Extract patient reference from encounter.subject
+  const patientReference = encounterResource?.subject as Reference<Patient> | undefined;
```

Comment:
Is this type cast necessary? Wouldn't that be the default type of `encounterResource?.subject`? We've been bitten by unsafe type casts a few times recently, so I'm extra wary around them these days...